src/libraries/UniswapV3Lib.sol

```diff
@@ -78,6 +113,54 @@ library UniswapV3Lib {
         );
     }
 
+    function addLiquidity(UniV3Context calldata context, AddLiquidityParams calldata params)
+        external
+        returns (uint256 tokenId, uint128 liquidity, uint256 amount0, uint256 amount1)
+    {
+        require(
+            params.amountDesired.amount0 > 0 || params.amountDesired.amount1 > 0,
+            "UniswapV3Lib/zero-amount"
+        );
+
+        require(params.maxSlippage > 0,     "UniswapV3Lib/max-slippage-not-set");
+        require(params.twapSecondsAgo != 0, "UniswapV3Lib/zero-twap-seconds");
+
+        IUniswapV3PoolLike pool = IUniswapV3PoolLike(context.pool);
+
+        AddLiquidityCache memory cache = _populateAddLiquidityCache(pool);
+
+        if (params.amountDesired.amount0 > 0) {
+            ERC20Lib.approve(context.proxy, cache.token0, address(params.positionManager), params.amountDesired.amount0);
+            context.rateLimits.triggerRateLimitDecrease(
+                RateLimitHelpers.makeAssetDestinationKey(context.rateLimitId, cache.token0, context.pool),
+                _scaleTo1e18(params.amountDesired.amount0, cache.token0Decimals)
+            );
+        }
+        if (params.amountDesired.amount1 > 0) {
+            ERC20Lib.approve(context.proxy, cache.token1, address(params.positionManager), params.amountDesired.amount1);
+            context.rateLimits.triggerRateLimitDecrease(
+                RateLimitHelpers.makeAssetDestinationKey(context.rateLimitId, cache.token1, context.pool),
+                _scaleTo1e18(params.amountDesired.amount1, cache.token1Decimals)
+            );
+        }
+
+        _validateAddLiquidityMinAmounts(context, params);
+
+        if (params.tokenId == 0) {
+            require(params.tick.lower >= params.tickBounds.lower, "UniswapV3Lib/invalid-tick-lower");
+            require(params.tick.upper <= params.tickBounds.upper, "UniswapV3Lib/invalid-tick-upper");
+
+            (tokenId, liquidity, amount0, amount1) = _mintLiquidity(context, params, cache);
+        } else {
+            require(params.positionManager.ownerOf(params.tokenId) == address(context.proxy), "UniswapV3Lib/proxy-does-not-own-token-id");
+
+            (liquidity, amount0, amount1) = _addLiquidityToExistingPosition(context, params);
+            tokenId = params.tokenId;
+        }
+
+        require(liquidity != 0, "UniswapV3Lib/no-liquidity-increased");
```

Comment:
I think we should double check that the amounts exceed the min amount requested to be safe here. We do it in the [DSS allocator deposit Univ3](https://github.com/sky-ecosystem/dss-allocator/blob/226584d3b179d98025497815adb4ea585ea0102d/src/funnels/DepositorUniV3.sol#L240-L248) as well.

Reply:
any guesses as to why, [given Uni also does this check](https://github.com/Uniswap/v3-periphery/blob/main/contracts/base/LiquidityManagement.sol#L88)?